### PR TITLE
Imti/add jest for existing tests

### PR DESCRIPTION
**Pull request by Imti**
## Motivation

Migrating from mocha -> jest for all tests

## Solution

Support jest for all packages

## PR Checklist

- [ ] Added Tests
- [ ] Updated Documentation
- [ ] Updated CHANGELOG.md for the appropriate package

### CODE REVIEW

**Review by xBalbinus — APPROVED** [review]

File: packages/multi-provider/tests/utils.test.ts [hunk 1]
```diff
@@ -18,44 +13,44 @@ describe('multi-provider utils', async () => {
...
-    expect(() => utils.canonizeId(tooLong)).to.throw('Too long');
-    expect(() => utils.canonizeId(badLen)).to.throw(
+    expect(cannonizedBytes32.toString()).toEqual(bytes32.toString());
+    expect(() => utils.canonizeId(tooLong)).toThrow('Too long');
+    expect(() => utils.canonizeId(badLen)).toThrow(
       'bad input, expect address or bytes32',
     );
   });
 
   it('converts Nomad Id to emv address', () => {
```
Comment (yourbuddyconner) [inline packages/multi-provider/tests/utils.test.ts:38]: emv -> evm

**Review by yourbuddyconner — APPROVED** [review]

File: packages/multi-provider/tests/index.test.ts [hunk 2]
```diff
@@ -25,78 +22,82 @@ describe('multi-provider', async () => {
...
     mp.registerDomain(chainADomain);
     mp.registerDomain(chainBDomain);
   });
 
   it('returns an array of domains', () => {
     const values = mp.registeredDomains;
-    expect(values).to.include(chainADomain);
-    expect(values).to.include(chainBDomain);
+    expect(values).toContainEqual(chainADomain);
+    expect(values).toContainEqual(chainBDomain);
```
Comment (ErinHales) [inline packages/multi-provider/tests/index.test.ts:34]: What's the difference between toContain and toContainEqual?

File: packages/multi-provider/tests/index.test.ts [hunk 3]
```diff
@@ -25,78 +22,82 @@ describe('multi-provider', async () => {
...
+    expect(() => mp.resolveDomainName('hi')).toThrow();
   });
 
   it('returns whether a given domain is registered', () => {
     const known = mp.knownDomain('a');
     const unknown = mp.knownDomain('c');
-    expect(known).to.be.true;
-    expect(unknown).to.be.false;
+    expect(known).toEqual(true);
+    expect(unknown).toEqual(false);
```
Comment (ErinHales) [inline packages/multi-provider/tests/index.test.ts:84]: Isn't there a toBeTrue method? And toBeFalse?

File: packages/multi-provider/tests/index.test.ts [hunk 4]
```diff
@@ -107,84 +108,92 @@ describe('multi-provider', async () => {
   // register A and B providers
   it('registers provider', async () => {
     // provider
-    expect(mp.providers.has('a')).to.be.false;
+    expect(mp.providers.has('a')).toEqual(false);
```
Comment (ErinHales) [inline packages/multi-provider/tests/index.test.ts:111]: Same here

File: packages/deploy/tests/index.test.ts [hunk 5]
```diff
@@ -14,7 +10,7 @@ describe('deploy', async () => {
       ctx.registerRpcProvider('rinkeby', 'http://dummyurl.com');
 
       const nomadCtx = ctx.asNomadContext;
-      expect(nomadCtx.getConnection('rinkeby')).to.not.be.undefined;
+      expect(nomadCtx.getConnection('rinkeby')).toBeDefined;
```
Comment (ErinHales) [inline packages/deploy/tests/index.test.ts:13]: Needs parenthesis?

Comment (ErinHales) [reply packages/deploy/tests/index.test.ts:13]: .toBeDefined()

File: packages/sdk-bridge/tests/index.test.ts [hunk 6]
```diff
@@ -24,23 +21,27 @@ describe('sdk-bridge', async () => {
           const bridge = context.getBridge(domain);
           const confBridge = conf.bridge[domain];
 
-          expect(bridge.bridgeRouter.address).to.equal(
+          expect(bridge).toBeDefined;
```
Comment (ErinHales) [inline packages/sdk-bridge/tests/index.test.ts:24]: Needs parenthesis?

File: packages/sdk-bridge/tests/index.test.ts [hunk 7]
```diff
@@ -24,23 +21,27 @@ describe('sdk-bridge', async () => {
...
+          expect(bridge?.bridgeRouter.address).toEqual(
             confBridge.bridgeRouter.proxy,
           );
-          expect(bridge.tokenRegistry.address).to.equal(
+          expect(bridge?.tokenRegistry.address).toEqual(
             confBridge.tokenRegistry.proxy,
           );
-          expect(bridge.ethHelper.address).to.equal(confBridge.ethHelper);
+
+          expect(bridge?.ethHelper).toBeDefined;
```
Comment (ErinHales) [inline packages/sdk-bridge/tests/index.test.ts:33]: Parenthesis?

File: packages/multi-provider/tests/index.test.ts (hunk 2 above)
Comment (Imti) [reply packages/multi-provider/tests/index.test.ts:34]: `toContain` is used for checking if a value in the array is strictly equal to the passed in value. This breaks down for objects depending on how the test is set up where you might not want strict equality (`===`). For this specific test, we really want to test the structure of the object, which we can use `toContainEqual` for.

File: packages/multi-provider/tests/index.test.ts (hunk 3 above)
Comment (Imti) [reply packages/multi-provider/tests/index.test.ts:84]: No, unfortunately jest doesn't have `toBeTrue` or `toBeFalse` methods. There's `toBeTruthy()` and `toBeFalsey()` but it works similar to how an if statement would evaluate a value to a boolean.

File: packages/sdk/tests/index.test.ts [hunk 8]
```diff
@@ -43,45 +42,46 @@ describe('NomadContext', () => {
...
           const replica = context.mustGetReplicaFor(homeDomain, remoteDomain);
-          expect(replica.address).to.equal(
+          expect(replica.address).toEqual(
             remoteConfCore.replicas[homeDomain].proxy,
           );
         }
 
         // returns undefined if no replica exists
-        expect(() => core.getReplica('none')).to.throw;
+        expect(() => core.getReplica('none')).toThrow;
```
Comment (ErinHales) [inline packages/sdk/tests/index.test.ts:67]: Parenthesis

**Review by ErinHales — APPROVED** [review]
Missed one spot but other than that looks good

**Pull request merged**